From 138bccda252b4f508afffd3662aa808ee6efe85f Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Thu, 7 Dec 2017 12:57:19 +0200
Subject: [PATCH 3052/5242] MLK-17140-1: drm: imx: dcss: add PM QoS

commit  d94d74f5f31841ba32a48512b95a71595f9fc4fa from
https://source.codeaurora.org/external/imx/linux-imx.git

PM QoS is needed so that cpuidle doesn not influence DCSS performance.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dcss/dcss-common.c |    9 +++++++++
 drivers/gpu/imx/dcss/dcss-prv.h    |    4 ++++
 2 files changed, 13 insertions(+)

diff --git a/drivers/gpu/imx/dcss/dcss-common.c b/drivers/gpu/imx/dcss/dcss-common.c
index 99a7057..28b43a9 100644
--- a/drivers/gpu/imx/dcss/dcss-common.c
+++ b/drivers/gpu/imx/dcss/dcss-common.c
@@ -19,6 +19,7 @@
 #include <linux/clk.h>
 #include <linux/pm_runtime.h>
 #include <linux/busfreq-imx.h>
+#include <linux/pm_qos.h>
 #include <video/imx-dcss.h>
 
 #include <drm/drm_fourcc.h>
@@ -473,6 +474,8 @@ static int dcss_suspend(struct device *dev)
 
 	dcss_clocks_enable(dcss, false);
 
+	pm_qos_remove_request(&dcss->pm_qos_req);
+
 	dcss_bus_freq(dcss, false);
 
 	return 0;
@@ -485,6 +488,8 @@ static int dcss_resume(struct device *dev)
 
 	dcss_bus_freq(dcss, true);
 
+	pm_qos_add_request(&dcss->pm_qos_req, PM_QOS_CPU_DMA_LATENCY, 0);
+
 	dcss_clocks_enable(dcss, true);
 
 	dcss_blkctl_cfg(dcss);
@@ -509,6 +514,8 @@ static int dcss_runtime_suspend(struct device *dev)
 
 	dcss_clocks_enable(dcss, false);
 
+	pm_qos_remove_request(&dcss->pm_qos_req);
+
 	dcss_bus_freq(dcss, false);
 
 	return 0;
@@ -521,6 +528,8 @@ static int dcss_runtime_resume(struct device *dev)
 
 	dcss_bus_freq(dcss, true);
 
+	pm_qos_add_request(&dcss->pm_qos_req, PM_QOS_CPU_DMA_LATENCY, 0);
+
 	dcss_clocks_enable(dcss, true);
 
 	dcss_blkctl_cfg(dcss);
diff --git a/drivers/gpu/imx/dcss/dcss-prv.h b/drivers/gpu/imx/dcss/dcss-prv.h
index 3df06c5..d2b739d 100644
--- a/drivers/gpu/imx/dcss/dcss-prv.h
+++ b/drivers/gpu/imx/dcss/dcss-prv.h
@@ -1,6 +1,8 @@
 #ifndef __DCSS_PRV_H__
 #define __DCSS_PRV_H__
 
+#include <linux/pm_qos.h>
+
 #define SET 0x04
 #define CLR 0x08
 #define TGL 0x0C
@@ -54,6 +56,8 @@ struct dcss_soc {
 
 	bool bus_freq_req;
 	bool clks_on;
+
+	struct pm_qos_request pm_qos_req;
 };
 
 /* BLKCTL */
-- 
1.7.9.5

